store/sqlstore/channel_store.go

```
@@ -3316,45 +3363,111 @@ func (s SqlChannelStore) buildFulltextClause(term string, searchColumns string)
+		fullClauseFmt := "((to_tsvector('english', %s)) @@ to_tsquery('english', ?))"
+
+		fulltextClause = sq.Expr(fmt.Sprintf(fullClauseFmt, cols), fulltextTerm)
+	} else if s.DriverName() == model.DatabaseDriverMysql {
+		splitTerm := strings.Fields(fulltextTerm)
+		for i, t := range strings.Fields(fulltextTerm) {
+			splitTerm[i] = "+" + t + "*"
+		}
+
+		fulltextTerm = strings.Join(splitTerm, " ")
+
+		fullClauseFmt := "MATCH(%s) AGAINST (? IN BOOLEAN MODE)"
+		fulltextClause = sq.Expr(fmt.Sprintf(fullClauseFmt, searchColumns), fulltextTerm)
+	}
+
+	return
+}
+
+func (s SqlChannelStore) performSearch(searchQuery sq.SelectBuilder, term string) (model.ChannelList, error) {
+	var sql string
+	var parameters []interface{}
+	sql, parameters, err := searchQuery.ToSql()
+	_ = parameters
+	_ = sql
+	if err != nil {
+		return model.ChannelList{}, errors.Wrapf(err, "failed to construct query with term='%s'", term)
 	}
 
 	var channels model.ChannelList
 
-	if _, err := s.GetReplica().Select(&channels, searchQuery, parameters); err != nil {
-		return nil, errors.Wrapf(err, "failed to find Channels with term='%s'", term)
+	err2 := s.GetReplicaX().Select(&channels, sql, parameters...)
+	if err2 != nil {
+		return model.ChannelList{}, errors.Wrapf(err2, "failed to find Channels with term='%s'", term)
+	}
+
+	if channels == nil {
+		return model.ChannelList{}, err2
 	}
 
 	return channels, nil
 }
 
-func (s SqlChannelStore) performGlobalSearch(searchQuery string, term string, parameters map[string]interface{}) (model.ChannelListWithTeamData, error) {
-	likeClause, likeTerm := s.buildLIKEClause(term, "c.Name, c.DisplayName, c.Purpose")
-	if likeTerm == "" {
-		// If the likeTerm is empty after preparing, then don't bother searching.
-		searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", "", 1)
+func (s SqlChannelStore) searchClause(term string) (searchQuery sq.Sqlizer) {
+	likeClause, err := s.squirrelBuildLIKEClause(term, "c.Name, c.DisplayName, c.Purpose")
+
+	_ = err
```

Please return the error upwards.